src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -2415,8 +2449,21 @@ static void continue_read_action_locked(grpc_chttp2_transport* t) {
   const bool urgent = !t->goaway_error.ok();
   GRPC_CLOSURE_INIT(&t->read_action_locked, read_action, t,
                     grpc_schedule_on_exec_ctx);
+  // Set min progress size on the read path iff we are able to advertise
+  // prefered rx crypto frame sizes to the peer and the peer has also indicated
+  // that it can adjust sending frame sizes. (The peer would have informed us of
+  // this cabability by sending prefered rx frame sizes to us). Otherwise set
+  // min_progress_size it to -1. The endpoints should ignore the
+  // min_progress_size value if set to -1.
+  int min_progress_size =
+      t->enable_preferred_rx_frame_advertisement &&
+              t->settings
+                  [GRPC_PEER_SETTINGS]
+                  [GRPC_CHTTP2_SETTINGS_GRPC_PREFERRED_RECEIVE_CRYPTO_FRAME_SIZE]
+          ? grpc_chttp2_min_read_progress_size(t)
+          : -1;
```

Comment:
Why -1 and not 1?

Reply:
in src/core/lib/secure_endpoint.cc, I'm using the -1 as a signal to avoid tcp auto sizing. The -1 tells the secure endpoint to ignore min progress sizes obtained b parsing the crypto frames and instead it will always pass 1 to the tcp layer.

Reply:
I removed the -1 now per our offline discussion. I also reverted all changes to secure_endpoint.cc